cpp/benchmarks/io/orc/orc_reader_benchmark.cpp

```diff
@@ -66,15 +66,130 @@ void ORC_read(benchmark::State& state)
+  return cudf_io::read_orc(read_options).metadata.column_names;
+}
+
+void BM_orc_read_varying_options(benchmark::State& state)
+{
+  auto state_idx        = 0;
+  auto const col_sel    = static_cast<column_selection>(state.range(state_idx++));
+  auto const row_sel    = static_cast<row_selection>(state.range(state_idx++));
+  auto const num_chunks = state.range(state_idx++);
+
+  auto const flags         = state.range(state_idx++);
+  auto const use_index     = (flags & 1) != 0;
+  auto const use_np_dtypes = (flags & 2) != 0;
+  auto const dec_as_float  = (flags & 4) != 0;
+  auto const ts_type       = cudf::data_type{static_cast<cudf::type_id>(state.range(state_idx++))};
+
+  auto const data_types =
+    dtypes_for_column_selection(get_type_or_group({int32_t(type_group_id::INTEGRAL_SIGNED),
+                                                   int32_t(type_group_id::FLOATING_POINT),
+                                                   int32_t(type_group_id::TIMESTAMP),
+                                                   int32_t(cudf::type_id::STRING)}),
+                                col_sel);
+  auto const tbl  = create_random_table(data_types, data_types.size(), table_size_bytes{data_size});
+  auto const view = tbl->view();
+
+  std::vector<char> orc_data;
+  cudf_io::orc_writer_options options =
+    cudf_io::orc_writer_options::builder(cudf_io::sink_info{&orc_data}, view);
+  cudf_io::write_orc(options);
+
+  auto const cols_to_read = select_column_names(get_col_names(orc_data), col_sel);
+  cudf_io::orc_reader_options read_options =
+    cudf_io::orc_reader_options::builder(cudf_io::source_info{orc_data.data(), orc_data.size()})
+      .columns(cols_to_read)
+      .use_index(use_index)
+      .use_np_dtypes(use_np_dtypes)
+      .timestamp_type(ts_type)
+      .decimals_as_float64(dec_as_float);
+
+  auto const num_stripes              = data_size / (64 << 20);
+  cudf::size_type const chunk_row_cnt = view.num_rows() / num_chunks;
+  for (auto _ : state) {
+    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
+
+    cudf::size_type rows_read = 0;
+    for (int32_t chunk = 0; chunk < num_chunks; ++chunk) {
+      auto const is_last_chunk = chunk == (num_chunks - 1);
+      switch (row_sel) {
+        case row_selection::ALL: break;
+        case row_selection::STRIPES: {
+          auto stripes_to_read = segments_in_chunk(num_stripes, num_chunks, chunk);
+          if (is_last_chunk) {
+            // Need to assume that an additional "overflow" stripe is present
+            stripes_to_read.push_back(num_stripes);
+          }
+          read_options.set_stripes(stripes_to_read);
+        } break;
+        case row_selection::NROWS:
+          read_options.set_skip_rows(chunk * chunk_row_cnt);
+          read_options.set_num_rows(chunk_row_cnt);
+          if (is_last_chunk) read_options.set_num_rows(-1);
+          break;
+        default: CUDF_FAIL("Unsupported row selection method");
+      }
```

Comment:
should these lines be part of timing measured?
if not, move the raii below these lines and wrap a `{}` block around raii & benchmarked code.
```c++
{
    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
    rows_read += cudf_io::read_orc(read_options).tbl->num_rows();
}
```

Reply:
I don't think I can instantiate timer in every iteration of the inner loop, AFAICT it should be done once per state.

Reply:
From https://github.com/google/benchmark/blob/master/README.md :
> When UseManualTime is used, the benchmarked code must call SetIterationTime once per iteration of the benchmark loop to report the manually measured time.

Reply:
not inside for-loop. Just a `{}` scope block. It's still instantiates timer once per iteration.
do you need the lines 118-137 to be part of the benchmarked code as well?

Reply:
I don't need those lines benchmarked, but I cannot place the timer in a scope with read_orc because it's in the loop and thus called multiple times in some cases.
The only option would be to make a vector of options and used them in the timed portion, but I think this would not measurably change the timing.